cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/sink/DBSink.java

```diff
@@ -111,7 +118,20 @@ public void initialize(BatchRuntimeContext context) throws Exception {
 
   @Override
   public void transform(StructuredRecord input, Emitter<KeyValue<DBRecord, NullWritable>> emitter) throws Exception {
-    emitter.emit(new KeyValue<DBRecord, NullWritable>(new DBRecord(input, columnTypes), null));
+    // Create StructuredRecord that only has the columns in this.columns
+    List<Schema.Field> outputFields = Lists.newArrayList();
+    for (String column : getColumns()) {
+      Schema.Field field = input.getSchema().getField(column);
+      Preconditions.checkNotNull(field, "Missing schema field for column '%s'", column);
+      outputFields.add(field);
+    }
+    StructuredRecord.Builder output = StructuredRecord.builder(
+      Schema.recordOf(input.getSchema().getRecordName(), outputFields));
+    for (String column : getColumns()) {
```

Comment:
This is useful for testing when we want to use a specific list of columns without having to go through unrelated code paths.